internal/command/jsonformat/structured/attribute_path/matcher.go

```diff
@@ -151,7 +151,7 @@ func (p *PathMatcher) GetChildWithKey(key string) Matcher {
 			continue
 		}
 
-		if path[0].(string) == key {
+		if p, ok := path[0].(string); ok && p == key {
```

Comment:
I think this might be just masking the problem elsewhere. If this is a float64 then it should have triggered the `GetChildWithIndex` function below. There's a type incompatibility happening somewhere, where a float64 is being saved into a place that Terraform thinks there should be a string.

I worry this will just mask the error and we'll never discover what is causing that initial incompatibility.